index.js

```diff
@@ -128,9 +162,12 @@ function parseChar(character, state) {
   } else if (character === ']') {
     state.squareDepth--;
   }
-  if (!state.blockComment && !state.lineComment && !wasComment) state.history = character + state.history;
-  state.lastChar = character; // store last character for ending block comments
-  return state;
+  function end() {
+    if (!state.blockComment && !state.lineComment && !wasComment) state.history = character + state.history;
+    state.lastChar = character; // store last character for ending block comments
+    return state;
+  }
+  return end();
```

Comment:
Why have you moved this functionality into `end` rather than just falling through to doing it after the `if` conditionals have all run?  I can't see an example of anywhere you don't do it?